trie/sync_bloom.go

```diff
@@ -110,12 +97,11 @@ func (b *SyncBloom) init(database ethdb.Iteratee) {
 		// If the database entry is a trie node, add it to the bloom
 		key := it.Key()
 		if len(key) == common.HashLength {
-			b.bloom.Add(syncBloomHasher(key))
+			b.bloom.AddHash(binary.BigEndian.Uint64(key))
 			bloomLoadMeter.Mark(1)
-		}
-		// If the database entry is a contract code, add it to the bloom
-		if ok, hash := rawdb.IsCodeKey(key); ok {
-			b.bloom.Add(syncBloomHasher(hash))
+		} else if ok, hash := rawdb.IsCodeKey(key); ok {
+			// If the database entry is a contract code, add it to the bloom
+			b.bloom.AddHash(binary.BigEndian.Uint64(hash))
```

Comment:
Any particular reason for calling binary.Bigendian to manually convert instead of `common.BytesToHash` like other occurances? Seems a bit dangerous as the two methods could go out of sync.

Reply:
Ah, you mean the `hashToUint64` ?